src/linesearches_compat.jl

```diff
@@ -0,0 +1,234 @@
+
+"""
+    LineSearchesStepsize <: Stepsize
+
+Wrapper for line searches available in the `LineSearches.jl` library.
+
+## Constructors
+
+    LineSearchesStepsize(
+        M::AbstractManifold,
+        ls,
+        is;
+        retraction_method::AbstractRetractionMethod=default_retraction_method(M),
+        vector_transport_method::AbstractVectorTransportMethod=default_vector_transport_method(M),
+    )
+    LineSearchesStepsize(
+        ls,
+        is;
+        retraction_method::AbstractRetractionMethod=default_retraction_method(M),
+        vector_transport_method::AbstractVectorTransportMethod=default_vector_transport_method(M),
+    )
+
+Wrap linesearch `ls` (for example `HagerZhang` or `MoreThuente`) and initial step selector
+`is` that will work on manifold `M`. Retraction used for determining the line along which
+the search is performed can be provided as `retraction_method`. Gradient vectors are
+transported between points using `vector_transport_method`.
+"""
+struct LineSearchesStepsize{
+    TLS,TIS,TRM<:AbstractRetractionMethod,TVTM<:AbstractVectorTransportMethod
+} <: Stepsize
+    linesearch::TLS
+    initial_step::TIS
+    retraction_method::TRM
+    vector_transport_method::TVTM
+end
+function LineSearchesStepsize(
+    M::AbstractManifold,
+    ls,
+    is;
+    retraction_method::AbstractRetractionMethod=default_retraction_method(M),
+    vector_transport_method::AbstractVectorTransportMethod=default_vector_transport_method(
+        M
+    ),
+)
+    return LineSearchesStepsize(
+        ls,
+        is;
+        retraction_method=retraction_method,
+        vector_transport_method=vector_transport_method,
+    )
+end
+function LineSearchesStepsize(
```

Comment:
At some point Manopt should decide on maybe skipping these, if the sell it as „we need M for defaults“

Reply:
The nice thing about this constructor is that we don't need a manifold to set non-default methods. Would you prefer to implement this differently?

Reply:
Ah you are right, the ones I did (with just kwargs) do `M=DefaultManifold()` but that is the same as having this one if there is more than just the Manifold-parameter. Then we can keep this for sure.